lib/active_shipping/carriers/ups.rb

```diff
@@ -998,6 +1000,10 @@ def response_message(document)
       [status, desc].select(&:present?).join(": ").presence || "UPS could not process the request."
     end
 
+    def rate_warning_messages(rate_xml)
+      rate_xml.xpath("RatedShipmentWarning").map { |warning| warning.text }
```

Comment:
It will return an empty set. I've gone ahead and added a new test case to cover this.